code/modules/projectiles/guns/energy/cell_loaded_vr/multi_cannon.dm

```diff
@@ -0,0 +1,69 @@
+//Why is this in here when it's not a subtype of cell_loaded? Because it has a similar function, and I couldn't be assed to find a better suited spot.
+
+/obj/item/weapon/gun/projectile/multi_cannon
+	name = "Curabitur Cannon"
+	
+	desc = "A cannon developed by Curabitur Scimed, this weapon incorporates both Vey-Med and precursor technology to create a medical alternative to chemicals on the field."
+	icon = 'icons/vore/custom_guns_vr.dmi'
+	icon_state = "healcannon"
+
+	description_info = "Created to fulfill the needs and wants of the Curabitur Scimed personell during rescue operations, the Healcannon is a marvel of reverse-engineering and utilization of unknown technologies. \
+	It makes use of the Vey-Med ML-3 'Medigun' microbattery technology, combining the effects of multiple 'microbatteries' into single 'macrobatteries' that are built around precursor void cores, \
+	allowing for the batteries to self-charge for prolonged field use. However, the weakened beams caused by the use of ununderstood technology created a need for a VERY strong focus, \
+	which lead to the Healcannon becoming a very bulky tool. Fortunately, it's not as heavy as it looks." 
+
+	w_class = ITEMSIZE_LARGE
+	load_method = SINGLE_CASING
+	max_shells = 1
+
+	handle_casings = HOLD_CASINGS
+	auto_eject = FALSE
+
+
+	caliber = "macrobat"
+
+
+	icon_override = 'icons/vore/custom_guns_vr.dmi'
+	item_state = "healcannon"
+
+/obj/item/weapon/gun/projectile/multi_cannon/update_icon()
+	. = ..()
+	overlays.Cut()
+	var/istate = "healcannon_0"
+	var/indicator_colour = null
+	if(chambered)
+		if(istype(chambered,/obj/item/ammo_casing/macrobattery)) //should never not happen. but. you never, never know with this damn, cursed game.
+			var/obj/item/ammo_casing/macrobattery/bat = chambered
+			indicator_colour = bat.bat_colour
+			if(bat.charge)
+				istate = "healcannon_20"
+				var/percent_charged = round((bat.charge/bat.max_charge)*100)
+				switch(percent_charged)
+					if(21 to 40)
+						istate = "healcannon_40"
+					if(41 to 60)
+						istate = "healcannon_60"
+					if(61 to 80)
+						istate = "healcannon_80"
+					if(81 to INFINITY)//gotta cover any admemes/other ways to get above max charge here.
+						istate = "healcannon_100"
+			else
+				indicator_colour = null			
+
+	var/image/x = image(icon = 'icons/vore/custom_guns_vr.dmi', icon_state = istate)
+	x.layer = FLOAT_LAYER
+	x.plane = FLOAT_PLANE
+	x.color = indicator_colour
+	overlays += x
```

Comment:
```suggestion
	var/image/x = image(icon = icon, icon_state = istate)
	x.color = indicator_colour
	add_overlay(x)
```
You can cut this down to this, and use SSoverlay to add the overlay (less lag-inducing, because it gets queued rather than forcing it to do it right then), as long as you implement the cut_overlays() change too.